go/ai/prompt.go

```diff
@@ -426,3 +430,174 @@ func renderDotprompt(templateText string, variables map[string]any, defaultInput
 	}
 	return str, nil
 }
+
+// LoadPromptFolder loads prompts and partials from the input directory for the given namespace
+func LoadPromptFolder(r *registry.Registry, dir string, namespace string) error {
+	useDefaultDir := false
+	if dir == "" {
+		dir = "./prompts"
+		useDefaultDir = true
+	}
+	promptsPath, err := filepath.Abs(dir)
+	if err != nil {
+		errMsg := fmt.Sprintf("Failed to resolve directory: %v", err)
+		if !useDefaultDir {
+			return errors.New(errMsg)
+		}
+		slog.Error(errMsg)
```

Comment:
```suggestion
		if !useDefaultDir {
			return fmt.Errorf("failed to resolve prompt directory %q: %w, dir, err)
		}
		slog.Debug("default prompt directory not found, skipping loading .prompt files", "dir", dir)
```

We probably don't want to be spamming error logs when the user doesn't even want to use this feature. The differentiation between default prompt dir and a manually set one is good.